### PR TITLE
Added Support for excluding IPs in nios_next_ip and created new pluggin for nios_next_network

### PR DESCRIPTION
**Pull request by justjais**
##### SUMMARY

Infoblox NIOS lookup plugins
Added support to exclude IPs from the results for nios_next_ip.py
Added support for network range search from network-container using nios_next_network.py

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

nios
##### ANSIBLE VERSION

```
2.7
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

**Comment by webknjaz** [comment]
Hi @justjais,

I believe that you've mistakenly marked this as a bug. Am I right?

Also, could you please add some integration (and maybe unit) testing to this PR?

-label needs_triage

**Comment by justjais** [comment]
@webknjaz Thanks for pointing out have added bug tag by mistake as it's a feature PR.
Also, just wanted to confirm that if we write integration/unit tests for plugin modules as for similar `nios` plugins there are no unit tests written as these plugins modules won't be used in any of the module integration tests. Kindly confirm.

File: lib/ansible/plugins/lookup/nios_next_ip.py [hunk 1]
```diff
@@ -81,10 +90,11 @@ def run(self, terms, variables=None, **kwargs):
             raise AnsibleError('unable to find network object %s' % network)
 
         num = kwargs.get('num', 1)
+        excludeIP = kwargs.get('exclude', [])
```
Comment (Qalthos) [inline lib/ansible/plugins/lookup/nios_next_ip.py]: would rather see this in snake_case unless there's a compelling reason not to

File: lib/ansible/plugins/lookup/nios_next_network.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            raise AnsibleError('missing CIDR argument in the form of xx')
+
+        provider = kwargs.pop('provider', {})
+        wapi = WapiLookup(provider)
+        network_obj = wapi.get_object('networkcontainer', {'network': network})
+
+        if network_obj is None:
+            raise AnsibleError('unable to find network-container object %s' % network)
+        num = kwargs.get('num', 1)
+        excludeIP = kwargs.get('exclude', [])
```
Comment (Qalthos) [inline lib/ansible/plugins/lookup/nios_next_network.py]: And again here

**Review by Qalthos — COMMENTED** [review]
Seems pretty straightforward

File: lib/ansible/plugins/lookup/nios_next_ip.py (hunk 1 above)
Comment (justjais) [reply lib/ansible/plugins/lookup/nios_next_ip.py]: ah.. coming from C++ background still leads me to such bummers. I'll do the change as `exclude_ip` and upload again.

File: lib/ansible/plugins/lookup/nios_next_network.py (hunk 2 above)
Comment (justjais) [reply lib/ansible/plugins/lookup/nios_next_network.py]: I'll update it to `exclude_ip`

**Pull request merged**